packages/client/src/Components/PaperEdits/PaperEdit/ProgramScript/PreviewCanvas/VideoContextProgressBar.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable template-curly-spacing */
+/* eslint-disable react/prop-types */
+import React from 'react';
+import styles from './VideoContextProgressBar.module.css';
+
+class VideoContextProgressBar extends React.PureComponent {
+  constructor(props) {
+    const {
+      width = 640,
+    } = props;
+
+    super(props);
+
+    this.width = width;
+    this.state = { progress: 0 };
+  }
+
+  componentDidUpdate() {
+    this.videoContext = this.props.videoContext;
+    this.setState({ duration: this.videoContext.duration });
+    this.updateProgress();
+  }
+
+  updateProgress = () => requestAnimationFrame(() => {
+    const currentTime = this.videoContext && this.videoContext.currentTime;
+    const progress = (currentTime / this.state.duration) * 100;
+
+    if (this.state.progress !== progress) this.setState({ progress });
+
+    requestAnimationFrame(this.updateProgress);
+  });
+
+  handleClick = ({ nativeEvent: { offsetX } }) => {
+    this.videoContext.currentTime =
+      (offsetX / this.width) * this.state.duration;
+  }
+
+  getTracks = () =>
+    this.state.duration && this.videoContext._sourceNodes.reverse().map(
+      ({ startTime, stopTime, elementURL }, i) => {
+        const marginLeft = (startTime / this.state.duration) * this.width;
+        const width =
+          ((stopTime - startTime) / this.state.duration) * this.width;
+        const key = `${elementURL.split('/').slice(-1).pop()}.${i}`;
+
+        return (
+          <div key={ key } style={ { pointerEvents: 'none' } } >
+            <div className={ styles.papercutsPlayerProgressTrackBuffer } />
+            <div className={ styles.papercutsPlayerProgressTrack }
+              style={ { width, marginLeft } }
+            />
+            <div className={ styles.papercutsPlayerProgressTrackBuffer }/>
+          </div>
+        );
+      });
+
+  render() {
+    if (!this.tracks) this.tracks = this.getTracks();
```

Comment:
This won't sync with any changes to the playlist.

If this matters (maybe it doesn't), you could replace `!this.tracks` with `this.tracks.lengths !== this.videoContext._sourceNodes.length`

Or just regen the track list on each render (but that would be 60 times a second)